### PR TITLE
[x64] Add test of jnp.arange() dtype corner case

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #8178

This adds a specific test of a current behavior of `jnp.arange` that is easy to break when mucking with JAX's default types, as in #8178. Rather than detecting this breakage via indirect downstream side-effects, it is more convenient to test it directly. I'm still uncertain whether this behavior is desirable, but having an explicit test will also help if we do decide to change this corner-case behavior.

### CODE REVIEW

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -5574,6 +5574,24 @@ def testArangeConcretizationError(self):
     with self.assertRaisesRegex(jax.core.ConcretizationTypeError, msg('stop')):
       jax.jit(lambda stop: jnp.arange(0, stop))(3)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": str(dtype), "dtype": dtype}
+    for dtype in [None] + float_dtypes))
+  def testArange64Bit(self, dtype):
+    # Test that jnp.arange uses 64-bit arithmetic to define its range, even if the
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply tests/lax_numpy_test.py:5581]: Can you clarify what you have in mind? e.g. would you compare `jnp.arange(*args, dtype=dtype)` to `np.arange(*args).astype(dtype)`?

[the remark this answers is not in the document]
Comment (jakevdp) [reply tests/lax_numpy_test.py:5581]: (The reason I focused on number of elements here is because this is how the error manifests downstream. Generally X32 outputs are expected to differ from X64 outputs in terms of precision, but if the number of outputs changes it causes problems)

[the remark this answers is not in the document]
Comment (hawkinsp) [reply tests/lax_numpy_test.py:5581]: Maybe it's fine like this. But please comment it a bit better to explain why the property you are testing for holds.

[the remark this answers is not in the document]
Comment (jakevdp) [reply tests/lax_numpy_test.py:5581]: OK, expanded the comment.
